File: leet_724_find_pivot_index___naive.c


int pivotIndex(int* nums, int numsSize)
{
    float   cur;
    int     sum, i;

    i = -1;
    sum = 0;
    while (++i < numsSize)
    {
        sum += nums[i];
    }
    cur = 0;
    i = -1;
    while (++i < numsSize)
    {
        if (i)
        {
            cur += nums[i - 1];
        }
        if (cur == (float) (sum - nums[i]) / 2)
        {
            return (i);
        }
    }
    return (-1);
}

File: leet_724_find_pivot_index___naive.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(n) {
    let sum = n.reduce((s, a) => s + a, 0);
    let cur = 0
    let i = -1
    while (++i < n.length){
        if (i)
            cur += n[i - 1]
        if (cur == (sum - n[i]) / 2)
            return i
    }
    return -1
};
